csrc/scheduler/registry.cpp

```diff
@@ -1969,12 +1954,46 @@ class PersistentKernelScheduler : public SchedulerEntry {
       return false;
     }
 
+    if (inner_reduction && outer_reduction) {
+      // get vectorize_factor, same process to that in getPersistentHeuristics
+      auto reduced_tv = ir_utils::getSoleProducerTv(reference_tv);
+      const auto vectorize_factor = vectorize_helper::getVectorizationFactor(
+          runtime_info,
+          reduced_tv,
+          data_cache,
+          (int)(reduced_tv->nDims() - properties.inner_most_dimension_ndims));
+      if (!checkCombinedReductionShape(
+              runtime_info, reduction_tvs, (int64_t)vectorize_factor)) {
+        scheduler_debug_utils::canScheduleRejectReason(
+            ScheduleHeuristic::Persistent,
+            "Inner dim of combined reduction should be a multiplication of a quarter warp and max vectorization factor!");
+        return false;
+      }
+
+      // check batches of persistent buffer
```

Comment:
Can you please add more detailed comments? I think it would be helpful to mention that:

- There's a preferred (minimum?) batch size, `batch_needed`, which is determined by such factors as ...
- There's also a maximum batch size, `batch_available`, which is determined by ...

Reply:
added comments.